duckduckgo/Android
# Site Permissions Settings

Task/Issue URL: https://app.asana.com/0/0/1202399817058486/f

### Description

Replace Location Settings screen with brand new Site Permissions screen where users will be able to manage microphone, camera and location permissions when sites request it.

### Steps to test this PR

_Disable ask for permission_
- Install from branch
- Go to Settings > Site Permissions
- Disable any permission (mic / camera / location)
- Go back to browser
- [ ] Check you are not asked for the disabled permission

_Websites list_
- Install from branch
- Allow mic or camera permissions for any website
- Go to Settings > Site Permissions
- [ ] Check website has been added to websites list

_Permissions per site_
- Install from branch
- Go to Settings > Site Permissions
- Tap on a website in "Manage Sites" list
- [ ] Check all permission settings are correct
- [ ] If main permission is disabled to ask -> setting should be set to "Deny"

_Change site permission setting_
- Install from branch
- Go to Settings > Site Permissions
- Tap on a website in "Manage Sites" list
- Tap on any permission to change the setting
- [ ] Check setting has been changed successfully

_Remove Site Permissions_
- Install from branch
- Go to Settings > Site Permissions
- Tap on a website in "Manage Sites" list
- Tap on toolbar menu > Remove
- [ ] Check website permissions have been removed

_Remove All Site Permissions_
- Install from branch
- Go to Settings > Site Permissions
- Tap on overflow menu in "Manage Sites" title
- Select "Remove All"
- [ ] Check all sites permissions have been removed

_Clear Site Permissions Data when Fire button is selected_
- Install from branch
- Make sure you have some site permissions saved
- Tap Fire button
- [ ] Check Site Permissions have been cleared

### UI changes
| Before  | After |
| ------ | ----- |
![Screenshot_20220905-122005_DuckDuckGo](https://user-images.githubusercontent.com/20798495/188440132-970d7880-7c98-46f5-97d2-b9f9320f5308.jpg)|![Screenshot_20220905-122047_DuckDuckGo](https://user-images.githubusercontent.com/20798495/188440110-89b367ec-bfd1-4fa1-a2f0-f629bd2a3a05.jpg)|
